src/lib/y2users/password_validator.rb

```diff
@@ -68,6 +69,20 @@ def issues
     # @return [Y2Users::User] user containing the password to validate
     attr_reader :user
 
+    LETTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".freeze
+    NUMBERS = "0123456789".freeze
+    # TODO: find source for this info beside bsc#175706
+    PASSWORD_REGEXP = /\A[-#{NUMBERS}#{LETTERS}!@#\:!\$%^&*() ,;:._+\/|?{}=\['\"`~<>\]\\]+\z/.freeze
+    def check_password
+      return _("No password entered.\nTry again.") if !content || content.empty?
+
+      return "" if content =~ PASSWORD_REGEXP
+
+      _("The password may only contain the following characters:\n" \
```

Comment:
This message needs to be adapted according to what we find about the non-allowed characters.
Something like

> The allowed characters are ... ... ...
> (which are the printable ASCII characters except ... and ...)

Reply:
One topic I would like to discuss on call is this unallowed characters as I cannot find reason for it beside not so easy to type so special ones. So maybe it should not be fatal and just warn user that password contain special or non-ascii characters that can be hard to type during login?

Reply:
Yes, basically this is to avoid a situation where the user puts non-ASCII characters in the password and then cannot log in at the console because for whatever reason the keyboard layout is wrong / defaults to en_US.
We actually did get such a bug report from a German user.